frame/dvm/ethereum/src/lib.rs

```diff
@@ -1076,9 +1080,9 @@ pub fn internal_transaction(nonce: U256, target: H160, input: Vec<u8>) -> Advanc
 		signature: ethereum::TransactionSignature::new(
 			// Reference https://github.com/ethereum/EIPs/issues/155
 			//
-			// But this transaction is sent by darwinia-issuing system from `0x0`
-			// So ignore signature checking, simply set `chain_id` to `1`
-			1 * 2 + 36,
+			// The internal transaction is sent by specific pallets, no signature
+			// validation, just create a valid transaction signature is enough.
+			T::ChainId::get() * 2 + 36,
 			H256::from_slice(&[55u8; 32]),
 			H256::from_slice(&[55u8; 32]),
```

Comment:
Any comment for this magic value ?

```
	pub fn new(v: u64, r: H256, s: H256) -> Option<Self> {
		const LOWER: H256 = H256([
			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
			0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
			0x00, 0x00, 0x00, 0x01,
		]);
		const UPPER: H256 = H256([
			0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
			0xff, 0xfe, 0xba, 0xae, 0xdc, 0xe6, 0xaf, 0x48, 0xa0, 0x3b, 0xbf, 0xd2, 0x5e, 0x8c,
			0xd0, 0x36, 0x41, 0x41,
		]);

		let v = TransactionRecoveryId(v);
		let is_valid = v.standard() <= 1 && r < UPPER && r >= LOWER && s < UPPER && s >= LOWER;

		if is_valid {
			Some(Self { v, r, s })
		} else {
			None
		}
	}
```

Reply:
I guess it's just a valid value between `LOWER` and `UPPER`.  No specific reason for this. @xiaoch05  Am I right?

Reply:
YES